tests/dialects/test_hw.py

```diff
@@ -25,3 +46,132 @@ def test_inner_sym_target():
     assert sub_target
     assert not sub_target.is_op_only()
     assert sub_target.is_field()
+
+
+@irdl_op_definition
+class ModuleOp(IRDLOperation):
+    name = "module"
+    region: Region = region_def()
+    sym_name = attr_def(StringAttr)
+    traits = frozenset({InnerSymbolTable(), SymbolOpInterface()})
+
+
+@irdl_op_definition
+class OutputOp(IRDLOperation):
+    name = "output"
+    traits = frozenset({IsTerminator()})
+
+
+@irdl_op_definition
+class CircuitOp(IRDLOperation):
+    name = "circuit"
+    region: Region | None = opt_region_def()
+    sym_name = attr_def(StringAttr)
+    traits = frozenset(
+        {InnerRefNamespace(), SymbolTable(), SingleBlockImplicitTerminator(OutputOp)}
+    )
+
+    def __post_init__(self):
+        for trait in self.get_traits_of_type(SingleBlockImplicitTerminator):
+            ensure_terminator(self, trait)
+
+
+def test_inner_symbol_table_interface():
+    """
+    Test that operations that conform to SymbolOpInterface have necessary attributes.
+    """
+
+    @irdl_op_definition
+    class MissingTraitModuleOp(IRDLOperation):
+        name = "module"
+        traits = frozenset({InnerSymbolTable()})
+
+    op0 = MissingTraitModuleOp()
+    iface0 = op0.get_trait(InnerSymbolTable)
```

Comment:
AFAIU you could hoist it out of the test function

Reply:
I was trying to scope the failing definitions to their test functions and only keeping the ones that are reused for several tests outside − not a pertinent distinction then I’m guessing?

Reply:
I think it's great as it is